src/Config/Stage/Deploy/SearchEngine.php

```diff
@@ -75,34 +85,73 @@ public function __construct(
      *
      * @return array
      */
-    public function get(): array
+    public function getConfig(): array
     {
-        $envSearchConfig = (array)$this->stageConfig->get(DeployInterface::VAR_SEARCH_CONFIGURATION);
-
-        if ($this->isSearchConfigValid($envSearchConfig)
-            && !$this->configMerger->isMergeRequired($envSearchConfig)
-        ) {
-            return $this->configMerger->clear($envSearchConfig);
+        if ($this->config === null) {
+            $resolveMerge = function () {
+                $searchConfig = (array)$this->stageConfig->get(DeployInterface::VAR_SEARCH_CONFIGURATION);
+
+                if (isset($searchConfig['engine']) && !$this->configMerger->isMergeRequired($searchConfig)) {
+                    return $this->configMerger->clear($searchConfig);
+                }
+
+                return $this->configMerger->mergeConfigs(
+                    $this->getSearchConfig(),
+                    $searchConfig
+                );
+            };
+
+            $this->config['system']['default']['catalog']['search'] = $resolveMerge();
```

Comment:
Why are we changing the configuration in a case when the client set-up `SEARCH_CONFIGURATION` without option merge require?
If someone set `SEARCH_CONFIGURATION`  to `mysql` and ElasticSuite is installed the code will force change it to ElasticSuite.

Reply:
Once ElasticSuite installed, other options will lead to exceptions and broken site

Reply:
Do we have a validator for this case? When a user tried to set custom engine which will be force changed to ElasticSuite?

Reply:
The last comment questions are not actual, validator exists `ElasticSuiteIntegrity`